Package request: libtorrent-rasterbar

<!-- Important note: motivating 'why this package is needed' will most likely give the package higher priority in the developers TODO-list -->

**Package description**
libtorrent is a feature complete C++ bittorrent implementation focusing on efficiency and scalability. It runs on embedded devices as well as desktops. This is the missing piece needed to run deluged on Termux.

**Link to home page and sources**
1. Home page: https://www.libtorrent.org/
2. Source code: https://github.com/arvidn/libtorrent

**Additional information**
Have you compiled or tried to compile the package on device?
* I'm having trouble getting the build environment container running on my MacBook and have not been able to try building this.

@FreddieOliveira 

> $ pkg install boost
> $ git clone https://github.com/arvidn/libtorrent.git
> $ mkdir libtorrent/build
> $ cd libtorrent/build
> $ **cmake -DCMAKE_INSTALL_PREFIX=/data/data/com.termux/files/usr ..**
> $ make -j8
> $ make install

Thank you for your instructions. I am having a problem at this step and don't know how to proceed:

$ cmake -DCMAKE_INSTALL_PREFIX=/data/data/com.termux/files/usr ..
-- The C compiler identification is Clang 11.0.0
-- The CXX compiler identification is Clang 11.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /data/data/com.termux/files/usr/bin/cc - skippe                                                                                                                                                             d
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /data/data/com.termux/files/usr/bin/c++ - ski                                                                                                                                                             pped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Found OpenSSL: /data/data/com.termux/files/usr/lib/libcrypto.so (found versio                                                                                                                                                             n "1.1.1i")
-- Found Boost: /data/data/com.termux/files/usr/lib/cmake/Boost-1.74.0/BoostConf                                                                                                                                                             ig.cmake (found version "1.74.0")
-- The following features have been enabled:

 * BUILD_SHARED_LIBS, build libtorrent as a shared library
 * dht, enable support for Mainline DHT
 * deprecated-functions, enable deprecated functions for backwards compatibility
 * encryption, Enables encryption in libtorrent
 * exceptions, build with exception support
 * extensions, Enables protocol extensions
 * i2p, build with I2P support
 * logging, build with logging
 * mutable-torrents, Enables mutable torrent support
 * streaming, Enables support for piece deadline

-- The following RECOMMENDED packages have been found:

 * OpenSSL, Full-strength general purpose cryptography library, <https://www.ope                                                                                                                                                             nssl.org/>
   Provides HTTPS support to libtorrent

-- The following REQUIRED packages have been found:

 * Threads
 * boost_headers (required version == 1.74.0)
 * Boost

-- The following features have been disabled:

 * static_runtime, build libtorrent with static runtime
 * build_tests, build tests
 * build_examples, build examples
 * build_tools, build tools
 * python-bindings, build python bindings
 * python-egg-info, generate python egg info
 * python-install-system-dir, Install python bindings to the system installation                                                                                                                                                              directory rather than the CMake installation prefix
 * gnutls, build using GnuTLS instead of OpenSSL

-- Configuring done
CMake Error at CMakeLists.txt:536 (add_library):
  Cannot find source file:

    deps/try_signal/try_signal

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .ispc


CMake Error at CMakeLists.txt:536 (add_library):
  No SOURCES given to target: torrent-rasterbar


CMake Generate step failed.  Build files cannot be regenerated correctly.

--- a/packages/libtorrent-rasterbar/build.sh
+++ b/packages/libtorrent-rasterbar/build.sh
@@ -1,0 +1,44 @@
+TERMUX_PKG_HOMEPAGE=https://libtorrent.org/
+TERMUX_PKG_DESCRIPTION="A feature complete C++ bittorrent implementation focusing on efficiency and scalability"
+TERMUX_PKG_LICENSE="BSD 3-Clause"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2.0.5
+TERMUX_PKG_SRCURL=https://github.com/arvidn/libtorrent/releases/download/v${TERMUX_PKG_VERSION}/libtorrent-rasterbar-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=e965c2e53170c61c0db3a2d898a61769cb7acd541bbf157cbbef97a185930ea5
+TERMUX_PKG_DEPENDS="boost, openssl, python"
+
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-Dboost-python-module-name=python
+-Dpython-bindings=ON
+"
+
+termux_step_pre_configure() {
+	_PYTHON_VERSION=$(. $TERMUX_SCRIPTDIR/packages/python/build.sh; echo $_MAJOR_VERSION)
+	LDFLAGS+=" -lpython${_PYTHON_VERSION}"
+}
+
+termux_step_create_debscripts() {
+	local pyext_orig="$TERMUX_PREFIX/lib/python${_PYTHON_VERSION%.*}/dist-packages/libtorrent.so"
+	local pyext_symlink="$TERMUX_PREFIX/lib/python${_PYTHON_VERSION}/site-packages/libtorrent.so"
+
+	echo "#!$TERMUX_PREFIX/bin/sh" > postinst
+	echo "pyext_orig=\"$pyext_orig\"" >> postinst
+	echo "pyext_symlink=\"$pyext_symlink\"" >> postinst
+	echo "mkdir -p \"\$(dirname \"\$pyext_symlink\")\"" >> postinst
+	echo "if [ ! -e \"\$pyext_symlink\" ]; then" >> postinst
+	echo "  ln -sf \"\$pyext_orig\" \"\$pyext_symlink\"" >> postinst
+	echo "fi" >> postinst
+	echo "exit 0" >> postinst
+	chmod 0755 postinst
+
+	echo "#!$TERMUX_PREFIX/bin/sh" > prerm
+	echo "pyext_orig=\"$pyext_orig\"" >> prerm
+	echo "pyext_symlink=\"$pyext_symlink\"" >> prerm
+	echo "if [ -L \"\$pyext_symlink\" ]; then" >> prerm
+	echo "  if [ \"\$(readlink \"\$pyext_symlink\")\" == \"\$pyext_orig\" ]; then" >> prerm
+	echo "    rm -f \"\$pyext_symlink\"" >> prerm
+	echo "  fi" >> prerm
+	echo "fi" >> prerm
+	echo "exit 0" >> prerm
+	chmod 0755 prerm
+}

